source/array_table_model.ts

```diff
@@ -1,33 +1,67 @@
 import * as Kola from 'kola-signals';
-import { Operation } from './operations';
-import { TableModel } from './table_model';
+import {AddRowOperation, MoveRowOperation, Operation, RemoveRowOperation,
+  Transaction, UpdateOperation} from './operations';
+import {TableModel} from './table_model';
 
 /** Implements a TableModel using an 2-dimensional array. */
 export class ArrayTableModel extends TableModel {
 
   /** Constructs an empty model. */
   constructor() {
     super();
-    this.dispatcher = new Kola.Dispatcher<Operation>();
+    this.dispatcher = new Kola.Dispatcher<Operation | Transaction>();
+    this.table = [];
+    this.transactionArray = [];
+    this.transactionDepth = -1;
   }
 
   /**
    * Marks the beginning of a transaction. In cases where a transaction is
    * already being processed, then the sub-transaction gets consolidated into
    * the parent transaction.
    */
-  public beginTransaction(): void {}
+  public beginTransaction(): void {
+    if(this.transactionDepth > -1) {
+      const lastNestedArray = this.getLastNestedArray(this.transactionArray,
+        this.transactionDepth);
+      lastNestedArray.push([]);
```

Comment:
What is the purpose of this nesting?

Reply:
I was under the impression that beginTransaction starts a new Transaction within the current one. I'll remove the nesting and change the logic to continue the existing transaction.